src/setup.rs

```diff
@@ -23,12 +23,18 @@ use std::{io, process::exit};
 use tokio_core::reactor::Handle;
 use tokio_signal::ctrl_c;
 
+// We need to allow these lints since the conditional compilation is
+// what makes them appear.
+#[allow(unused_must_use, unreachable_code, unused_variables)]
```

Comment:
At least with the changes below and on Rust 1.63, none of those should be necessary any more.